pkg/services/internal/whatsonchain/is_valid_root.go

```diff
@@ -0,0 +1,77 @@
+package whatsonchain
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/bsv-blockchain/go-sdk/chainhash"
+)
+
+// IsValidRootForHeight checks if the provided Merkle root is valid for the given block height.
+func (woc *WhatsOnChain) IsValidRootForHeight(ctx context.Context, root *chainhash.Hash, height uint32) (bool, error) {
+	if cached, ok := woc.rootCache[height]; ok {
+		return cached.IsEqual(root), nil
+	}
+
+	var lastErr error
+
+	for range woc.rootForHeightValidationRetries {
+		attemptCtx := ctx
+		var cancel context.CancelFunc
+		if woc.rootForHeightValidationTimeout > 0 {
+			attemptCtx, cancel = context.WithTimeout(ctx, woc.rootForHeightValidationTimeout)
+			defer cancel()
+		}
+		ok, err := woc.fetchAndCompare(attemptCtx, root, height)
+		if err == nil {
+			return ok, nil
+		}
+
+		if ce := attemptCtx.Err(); ce != nil {
+			return false, fmt.Errorf("context canceled while validating Merkle root for height %d: %w", height, ce)
+		}
+
+		lastErr = err
+	}
```

Comment:
Sorry, but the `woc.rootForHeightValidationTimeout` is not a timeout value for **HTTP request** but rather a `delay` between retries:

```suggestion
		for range woc.rootForHeightValidationRetries {
		ok, err := woc.fetchAndCompare(ctx, root, height)
		if err == nil {
			return ok, nil
		}
		
		select {
		case <-ctx.Done():
			return false, fmt.Errorf("context canceled while validating Merkle root for height %d: %w", height, ctx.Err())
		case <-time.After(woc.rootForHeightValidationTimeout):
			// continue to retry
		}

		lastErr = err
	}
```

I also suggest to rename the `rootForHeightValidationTimeout` and the matching config param.